bin/add-exercise

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if (( $# != 1 )); then
+    echo >&2 "Populates a new directory for a practice exercise."
+    echo >&2 "Usage: $0 <exercise-slug>"
+    exit 1
+fi
+
+die() { echo >&2 "$*"; exit 1; }
+
+required_tool() {
+    command -v "$1" >/dev/null 2>&1 ||
+        die "$1 is required but not installed. Please install it and make sure it's in your PATH."
+}
+required_tool jq
+required_tool curl
+
+[[ -f ./bin/fetch-configlet ]] || die "run this script from the repo's root directory."
+
+slug="${1}"
+name=$(echo "${slug}" | sed 's/\b\w/\u&/g; s/-/ /g')    # assumes GNU sed
```

Comment:
```suggestion
name=$(sed 's/\b\w/\u&/g; s/-/ /g' <<< "${slug}")    # assumes GNU sed

```
Or `name=${slug//-/ }`. What does the first part of the sed do?

Reply:
This is a half-assed command to transform `pop-count` to `Pop Count`

The missing half-ass is for exercises like `etl` -> `Etl` (should be `ETL`)

Reply:
Can be done in bash:
```bash
$ slug='pop-count'
$ IFS=- read -ra words <<<"$slug"
$ name=${words[*]^}
 declare -p slug name
declare -- slug="pop-count"
declare -- name="Pop Count"
```

Reply:
Alternately:
```bash
name=${slug^}
while [[ $name =~ (.*)-([[:lower:]])(.*) ]]; do
  name="${BASH_REMATCH[1]} ${BASH_REMATCH[2]^}${BASH_REMATCH[3]}"
done
```